pkg/sql/tablewriter.go

```diff
@@ -794,16 +800,42 @@ func (td *tableDeleter) deleteAllRowsFast(
 			EndKey: tablePrefix.PrefixEnd(),
 		}
 	}
-	log.VEventf(ctx, 2, "DelRange %s - %s", resume.Key, resume.EndKey)
-	td.b.DelRange(resume.Key, resume.EndKey, false /* returnKeys */)
-	td.b.Header.MaxSpanRequestKeys = limit
+	// The most efficient deletion can be done using the ClearRange
+	// method. However, to maintain backwards compatibility, if no GC
+	// deadline is set for the table, we must revert to using
+	// DeleteRange. This is necessary because a legacy DROP or TRUNCATE
+	// would not set the GC deadline, but would still expect the data
+	// to remain historically queryable until the zone's TTL expires.
+	if td.tableDesc().GCDeadline == 0 {
+		log.VEventf(ctx, 2, "backwards-compatible DelRange %s - %s", resume.Key, resume.EndKey)
```

Comment:
Not sure the string "backwards-compatible" is needed / useful.
At any rate you need to update the condition in `sql/show_trace.go` (`(p *planner) ShowTrace`) accordingly.

Reply:
Reverted the "backwards-compatible" bit.